src/backend/utils/misc/queryenvironment.c

```diff
@@ -86,9 +95,7 @@ create_queryEnv2(MemoryContext cxt, bool top_level)
 
 	if (top_level) {
 		queryEnv = topLevelQueryEnv;
-		queryEnv->namedRelList = NIL;
-		queryEnv->dropped_namedRelList = NIL;
-		queryEnv->savedCatcacheMessages = NIL;
+		tsql_setup_queryEnv(queryEnv);
```

Comment:
Hooks can help us reduce regression chances. We should never call a TSQL function in APG instance. I think it is worth having hooks so that we never execute these code path outside BBF env.

Reply:
I generally agree that we shouldn't be calling tsql functions in APG code. The problem with this one in particular though, is that the members of QueryEnvironment are not visible outside the file, so we would have to make the queryenv struct public to be able to move that into a hook. An alternative would be making `tsql_setup_queryEnv` a public function that we invoke from a hook in the extension, but this to me seems counterintuitive to have a public function for private struct invoked through a hook. 

Other types of checks like sql_dialect might not be sufficient either because we can enter psql dialect in Babelfish code. 

At least for this case, I think we can add a comment saying to not introduce any code here that may have side effects given that the function could be called from APG as well.

Reply:
agreed to Surendra here btw.